esrally/driver/runner.py

```diff
@@ -912,12 +955,38 @@ async def scroll_query(es, params):
                 "took": took
             }
 
-        if params.get("use-search-after"):
-            return await search_after_query(es, params)
-        elif "pages" in params:
-            return await scroll_query(es, params)
+        def _extract_search_after_properties(response: BytesIO, get_point_in_time: bool, hits_total: int) -> (dict, List):
+            properties = ["timed_out", "took"]
+            if get_point_in_time:
+                properties.append("pit_id")
+            # we only need to parse these the first time
+            if hits_total is None:
+                properties.extend(["hits.total", "hits.total.value", "hits.total.relation"])
+
+            parsed = parse(response, properties)
+
+            # standardize these before returning...
+            parsed["hits.total.value"] = parsed.pop("hits.total.value", parsed.pop("hits.total", 0))
+            parsed["hits.total.relation"] = parsed.get("hits.total.relation", "eq")
+
+            response_str = response.getvalue().decode("UTF-8")
+            index_of_last_sort = response_str.rfind('"sort"')
+            sort_pattern = r"sort\":([^\]]*])"
```

Comment:
Can we turn this into a class variable and compile it with `re.compile`? Can you please also add documentation what's the intent behind this regex and how to read it so it's easier to maintain this in the future? 

I think it would also be great to extract `_extract_search_after_properties` into a top-level function and add dedicated tests for it. This would also make it easier to understand the possible inputs and cases that you're covering here.

If you do that, then you need to somehow reference the compiled regex, so you could implement it as a small class. I'm thinking about something along the lines of this:

```python
class SearchAfterExtractor:
    sort_pattern = re.compile(...)

    def __call__(response: BytesIO, get_point_in_time: bool, hits_total: int) -> (dict, List):
        ...

# reference this then in the Query runner via:
class Query:
    def __init__(...):
        self.extract_search_after_properties = SearchAfterExtractor()
```

Reply:
great idea